src/hooks/useSwitchWallet.ts

```diff
@@ -1,25 +1,25 @@
-import { useDispatch } from 'react-redux';
-import { addressSetSelected, walletsSetSelected } from '../redux/wallets';
-import useInitializeWallet from './useInitializeWallet';
-import { toChecksumAddress } from '@/handlers/web3';
+import { ensureChecksumAddress } from '@/handlers/web3';
 import { RainbowAccount } from '@/model/wallet';
-import useWallets from './useWallets';
+import { useWalletsStore } from '@/state/wallets/walletsStore';
+import useInitializeWallet from './useInitializeWallet';
 
 export default function useSwitchWallet() {
   const initializeWallet = useInitializeWallet();
-  const dispatch = useDispatch();
-  const { wallets } = useWallets();
+  const wallets = useWalletsStore(state => state.wallets);
 
   const switchToWalletWithAddress = async (address: string): Promise<string | null> => {
-    const walletKey = Object.keys(wallets!).find(key => {
+    if (!wallets) return null;
+
+    const walletKey = Object.keys(wallets).find(key => {
       // Addresses
-      return wallets![key].addresses.find((account: RainbowAccount) => account.address.toLowerCase() === address.toLowerCase());
+      return wallets[key].addresses.find((account: RainbowAccount) => account.address.toLowerCase() === address.toLowerCase());
     });
 
     if (!walletKey) return null;
-    const p1 = dispatch(walletsSetSelected(wallets![walletKey]));
-    const p2 = dispatch(addressSetSelected(toChecksumAddress(address)!));
-    await Promise.all([p1, p2]);
+
+    const { setSelectedWallet } = useWalletsStore.getState();
+    setSelectedWallet(wallets[walletKey], ensureChecksumAddress(address));
```

Comment:
the pattern elsewhere in the function seems to be to return `null` for unexpected scenarios - now with using `ensureChecksumAddress` instead of `toChecksumAddress`, the function could throw from the underlying `getAddress` function; it's good that we aren't setting the potential `null` value from `toChecksumAddress` anymore, but I don't think we're set up to handle a potential thrown error upstream as opposed to the `null` value